### PR TITLE
Fix incorrect syntax errors for MSSQL driver

### PR DESCRIPTION
**Pull request by zhevron**
With the MSSQL driver recently merged in, I'm getting errors when running against my MSSQL 2017 instance:
```
try lock failed in line 0: EXEC sp_getapplock @Resource = ?, @LockMode = 'Update', @LockOwner = 'Session', @LockTimeout = 0 (details: mssql: Incorrect syntax near '?'.)
```

MSSQL uses `@pN` syntax for parameters instead of `?`.

### CODE REVIEW

**Comment by zhevron** [comment]
My poor Windows PC with docker-machine isn't too fond of running the tests. I'll fire up a VM and look into it!

**Comment by zhevron** [comment]
The existing tests will catch regressions here. They were passing because the connection was opened with the deprecated 'mssql' driver instead of the 'sqlserver' one. The 'mssql' driver still uses `?` parameters and is not recommended to use.

**Comment by zhevron** [comment]
@dhui You're right. Because of `WithInstance`, we can't force the end user to use the `sqlserver` driver. As far as I can tell, that leaves us with two options:

1) Add support for both `sqlserver` and `mssql` drivers with different SQL statements. I'm thinking config option.
2) Add a notice that only the `sqlserver` driver is supported. `mssql` is deprecated already anyway. This would prevent legacy applications from adopting the library though.

Which would you think is better? I'm leaning towards option 1 currently. Can always remove the support once the upstream driver removes their support.

**Comment by zhevron** [comment]
Not sure what's going on with the tests here. Repeatedly seeing this error on unrelated tests:
```
dktest.go:54: Error parsing image pull response: context deadline exceeded
```
followed by
```
dktest.go:182: Failed to run image: mcr.microsoft.com/mssql/server:2019-latest error: Error: No such image: mcr.microsoft.com/mssql/server:2019-latest
```
for all the database images.

**Comment by dhui** [comment]
Looks like MCR or TravisCI were being slow.
We could up the image pull timeout to 5m in `mssql_test.go`.

**Comment by dhui** [comment]
@zhevron I'm also thinking that we should rename the driver to `sqlserver` to also reduce confusion. Thoughts?

**Comment by zhevron** [comment]
@dhui I think it's clearer which vendor it is with the driver name remaining as `mssql`, but I can also see the case for renaming it since that's what the upstream driver did. Since the `sqlserver` name is the future for that library, I would agree a name change here is warranted. I'll get it done.

As for the tests, it was happening to several of the database drivers, so increasing the timeout in `mssql_test.go` won't do much. I saw it with `postgres` and several others too. Let's just see if whoever was experiencing problems sorts it out first.

File: database/sqlserver/sqlserver.go [hunk 1]
```diff
@@ -44,7 +43,7 @@ type Config struct {
 }
 
 // MSSQL connection
```
Comment (dhui) [inline database/sqlserver/sqlserver.go]: Comment needs to be updated as well

**Pull request merged**